pkg/network/tracer/gateway_lookup.go

```diff
@@ -153,7 +154,12 @@ func (g *gatewayLookup) Lookup(cs *network.ConnectionStats) *network.Via {
 
 				// cache an empty result so that we don't keep hitting the
 				// ec2 metadata endpoint for this interface
-				g.subnetCache.Add(r.IfIndex, nil)
+				if errors.IsTimeout(err) {
+					// retry after a minute if we timed out
+					g.subnetCache.Add(r.IfIndex, time.Now().Add(time.Minute))
```

Comment:
we could cheaply add a tag to `subnetLookupErrors` so that we know how often the timeout case is hit